### PR TITLE
persistent-mysql instrumentation

### PR DESCRIPTION
**Pull request by kakkun61**
- new instrumentation for persistent-mysql
- add `wrapSqlBackend'` to `OpenTelemetry.Instrumentation.Persistent`

### CODE REVIEW

File: instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+createMySQLPool ::
+  (MonadUnliftIO m, MonadLoggerIO m) =>
+  Otel.TracerProvider ->
+  -- | Attributes that are specific to providers like MySQL, PostgreSQL, etc.
+  [(Text, Otel.Attribute)] ->
+  -- | Connection information.
+  MySQL.ConnectInfo ->
+  -- | Number of connections to be kept open in the pool.
+  Int ->
+  m (Pool SqlBackend)
```
Comment (iand675) [inline instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs:52]: I know that `hs-opentelemetry-instrumentation-persistent` currently only supports `SqlBackend`, but I wonder if we could add support for `IsSqlBackend backend` as part of this PR?

File: instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  pure (conn, backend')
+
+
+{- | Same as 'withMySQLPool', but instead of opening a pool
+of connections, only one connection is opened.
+-}
+withMySQLConn ::
+  (MonadUnliftIO m, MonadLoggerIO m) =>
+  Otel.TracerProvider ->
+  -- | Attributes that are specific to providers like MySQL, PostgreSQL, etc.
```
Comment (iand675) [inline instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs:98]: I don't really see where any attributes are being added for MySQL connections in this PR. Is this expected to be a manual addition by applications?

File: instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs (hunk 1 above)
Comment (kakkun61) [reply instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs:52]: How should I change after all?

File: instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs (hunk 2 above)
Comment (kakkun61) [reply instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs:98]: No extra attributes are added by hs-opentelemetry-instrumentation-persistent-mysql now.

Ah, I found this document now. https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/database/

**Comment by kakkun61** [comment]
That CI failure is caused by https://github.com/haskell/actions/issues/187.

File: instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs (hunk 2 above)
Comment (kakkun61) [reply instrumentation/persistent-mysql/src/OpenTelemetry/Instrumentation/Persistent/MySQL.hs]: I made some attributes be added at https://github.com/iand675/hs-opentelemetry/pull/50/commits/3fcbf3e863c8bba8aef2edd545261d9e5fa28a0a.

**Comment by kakkun61** [comment]
marge #53 first

**Comment by kakkun61** [comment]
@iand675 How about this PR?

**Review by iand675 — APPROVED** [review]

**Pull request merged**